### PR TITLE
feat(cactus-fwk): Add error boundary component

### PR DESCRIPTION
**Pull request by Dhalton**
CACTUS-39

### CODE REVIEW

**Comment by JamesNimlos** [comment]
Let's add a `withErrorBoundary` higher order component for them to use also

**Comment by Dhalton** [comment]
Closing until I can refactor into a HOC

File: modules/cactus-fwk/src/ErrorBoundary.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      return <Fragment>{children}</Fragment>
+    }
+  }
+}
+
+const noop = (error: Error, info: React.ErrorInfo) => {}
+
+const withErrorBoundary = <BaseProps extends any>(
+  BaseComponent: ComponentType<BaseProps>,
+  onError?: OnError,
```
Comment (JamesNimlos) [inline modules/cactus-fwk/src/ErrorBoundary.tsx]: What do you think about making this mandatory and throwing if you don't provide an `onError` here instead?

File: modules/cactus-fwk/src/ErrorBoundary.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+interface ErrorBoundaryState {
+  error: Error | undefined
+  errorInfo: ErrorInfo | undefined
+}
+
+const initialState: ErrorBoundaryState = { error: undefined, errorInfo: undefined }
+
+export class ErrorBoundary extends React.Component<ErrorBoundaryProps, ErrorBoundaryState> {
+  readonly state: ErrorBoundaryState = initialState
```
Comment (JamesNimlos) [inline modules/cactus-fwk/src/ErrorBoundary.tsx]: We could also potentially throw if they don't provide an `onError` onto the component as well to ensure developers are adding it? Or we could start adding propTypes definitions now?

**Pull request merged**